src/calliope/preprocess/data_sources.py

```diff
@@ -23,7 +27,7 @@
 
 LOGGER = logging.getLogger(__name__)
 
-DTYPE_OPTIONS = {"str": str, "float": float}
+locale.setlocale(locale.LC_ALL, "en_GB.utf-8")
```

Comment:
Contrary to the PR description, the locale is hardcoded. I think that's a bad idea and I think it might break on some platforms, e.g. Windows, where locale strings have a different format. Can we use a `locale` module independent approach to converting strings to numbers?

Reply:
Reverted to using system locale